src/structures/GuildChannel.js

```diff
@@ -254,19 +261,23 @@ class GuildChannel extends Channel {
    *   .then(channel => console.log(channel.permissionOverwrites.get(message.author.id)))
    *   .catch(console.error);
    */
-  createOverwrite(userOrRole, options, reason) {
-    userOrRole = this.guild.roles.resolve(userOrRole) || this.client.users.resolve(userOrRole);
-    if (!userOrRole) return Promise.reject(new TypeError('INVALID_TYPE', 'parameter', 'User nor a Role'));
-
-    const type = userOrRole instanceof Role ? OverwriteTypes.role : OverwriteTypes.member;
+  createOverwrite(userOrRole, options, extraInfo = {}) {
+    let userOrRoleID = this.guild.roles.resolveID(userOrRole) || this.client.users.resolveID(userOrRole);
+    let { type, reason } = extraInfo;
+    if (!(type === 0 || type === 1)) {
```

Comment:
```suggestion
    if (typeof type !== 'number') {
```

Reply:
This will let any number to pass which will result in an API error that only value 0 or 1 is allowed.

Reply:
In case that happens, Discord will just throw an error.

Similarly, if Discord were to add more types, this method wouldn't need changes (as long as we provide the ID and the new type), so it's future-proofed.